keps/sig-storage/3751-volume-provisioned-io/README.md

```diff
@@ -0,0 +1,473 @@
+
+## Motivation
+
+
+### Goals
+
+* Cloud-provider independent specification of basic QoS parameters
+* Specifying QoS at volume creation
+* Changing QoS parameters on-line for existing volumes
+* Specify quotas in a similar way to that of volume capacity
+
+### Non-Goals
+
+* OS-level IO QoS, for example cgroup policy
+* Inter-pod QoS. This KEP is restricted to what QoS can be defined on a volume.
+* Volume QoS classes. This proposal is for fine-grained provisioned IO control, and not for defining general classes of volume performance. Currently, storage vendors may accomplish this by adding custom parameters to a storage class. A future KEP may codify such behavior with common storage class parameter names, but that is out of scope of this KEP.
+* Scheduling based on total IO limits of a node. For example, total throughput for many cloud provider network attached storage volumes is limited by instance network bandwidth and may prevent achieving the provisioned IO. Scheduling to mitigate these factors is out of scope but it should be considered as a future extension of this KEP.
+
+## Proposal
+
+### Kubernetes API
+Add two well-known ResourceName keys to the existing ResourceList type used in PersistentVolumeClaim.Spec.Resources.{Limits,Requests} and PersistentVolume.Spec.Capacity:
+* ProvisionedIOPS
+* ProvisionedThroughput
+These values can be changed on a PersistentVolumeClaim, and will be reconciled against the storage provider in the same way as storage capacity resizing. The status of a PVC will reflect the currently provisioned IO as reported by the storage provider (in the same way that the status states the current size of the volume during a resize operation).
+
+```
+// Example:
+kind: PersistentVolumeClaim
+...
+spec:
+  ...
+  resources:
+requests:
+  storage: 4Gi
+  throughput: "1.2M"
+  iops: "160000"
+// Vendors may add additional resources using Device Plugin syntax
+// (azure.io/tunable: “100”)
+```
+
+Add PVC Status field; change ResizeStatus alpha field to AllocatedResourceStatus map.
```

Comment:
Can you explain API changes being proposed to `resizeStatus` field in more detail?

Reply:
This is added in the design details.